src/util/config.rs

```diff
@@ -113,8 +120,60 @@ pub fn parse_readable_int(size: &str) -> Result<i64, ParseConfigError> {
         "m" => Ok((num * (MINTUE as f64)) as i64),
         "h" => Ok((num * (HOUR as f64)) as i64),
 
-        _ => Err(ParseConfigError::ReadableNumber),
+        _ => Err(ConfigError::ReadableNumber),
+    }
+}
+
+/// `validate_addrs` validates a list of addresses. Addresses are formed like "Host:Port".
+/// More details about **Host** and **Port** can be found in RFC3986 section 3.2.2 and 3.2.3.
+pub fn validate_addrs(addrs: &[&str]) -> Result<(), ConfigError> {
+    fn new_err(addr: &str) -> ConfigError {
+        ConfigError::PDList(format!("invalid addr: {}", addr))
+    }
+
+    for addr in addrs {
+        if let (Err(_), Err(_)) = (SocketAddrV4::from_str(addr), SocketAddrV6::from_str(addr)) {
+            let parts: Vec<&str> = addr.split(':')
+                .map(|s| s.trim())
+                .filter(|s| !s.is_empty())
+                .collect();
+
+            // 2 for ["Host", "Port"]
+            if parts.len() != 2 {
+                return Err(new_err(addr));
+            }
+
+            // Check Port.
+            let port: u16 = try!(parts[1]
+                .parse()
+                .map_err(|_| new_err(addr)));
+            // Port = 0 is invalid.
+            if port == 0 {
+                return Err(new_err(addr));
+            }
+
+            // Check Host.
+            for c in parts[0].chars() {
```

Comment:
I think it is still a little complex here. We can reference Go https://golang.org/pkg/net/#SplitHostPort, https://golang.org/src/net/ipsock.go

For host:port, it only check whether it can be split into two parts with ":".